classes/reminders_repository.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace local_augmented_teacher;
+
+class reminders_repository
+{
+
+    private \moodle_database $db;
+
+    public function __construct(\moodle_database $DB) {
+        $this->db = $DB;
+    }
+
+    /**
+     * Returns activity reminders
+     * @return \moodle_recordset
+     * @throws \dml_exception
+     */
+    public function get_activity_reminders(): \moodle_recordset {
+        $sql = "SELECT rem.*, cm.completionexpected,
+                   case rem.type
+                       when ".REMINDER_BEFORE_DUE." then cm.completionexpected - rem.timeinterval
+                       when ".REMINDER_AFTER_DUE." then cm.completionexpected + rem.timeinterval
+                   end as scheduled
+              FROM {local_augmented_teacher_rem} rem
+              JOIN {course_modules} cm
+                ON rem.cmid = cm.id
+             WHERE rem.deleted = ?
+               AND rem.enabled = ?
+               AND rem.messagetype = ?";
+        $sql_params = array(0, 1, MESAGE_TYPE_REMINDER);
+
+        return $this->db->get_recordset_sql($sql, $sql_params);
```

Comment:
It doesn't seem like you are using the recordset for anything. You could just call get_records_sql. Then you won't need to close the connection outside the method. Applies to all the places where you are calling get_recordset_sql.